### PR TITLE
feat(app): 2791 - New mig page.

### PR DESCRIPTION
**Pull request by marin42**
**Pull Request Title Rules**

Pull request format must respect this format:

```
<type>(<scope>, <scope>...): <Notion ID> or "sentry" - <description>
```

Allowed types: `feat`, `fix`, `refactor`, `chore`.
Allowed scopes: `release`, `app`, `admin`, `misc`, `api`, `all`, `github`, `terraform`, `kb`, `lib`.

Notion ID, followed by a `-` is **mandatory** on types `refactor`, `feat` and `fix`.

_If you really need to bypass the pull request title validation, you can add the `bypass title` label to you PR._

Examples of valid PR titles:

- `chore(release): New version 1.3.3.7`
- `refactor(api): 1789 - Removed monarchy from codebase`
- `fix(api, app, admin): 1981 - User cannot delete each other accounts anymore.`
- `fix(api): sentry - Implement capture or fix a bug.`

**Description**

**Todo**

**Checklist**

- [ ] **⚠️ My code can have side-effects on other part of the code-base**
- [ ] I have added the Plausible tags/events
- [ ] I have performed a self-review of my code (and removed console.log)

**Ticket / Issue**

**Testing instructions**

### CODE REVIEW

File: app/src/scenes/militaryPreparation/components/DocumentsPM.jsx [hunk 1]
```diff
@@ -7,28 +7,28 @@ import FileCard from "./FileCard";
...
+    REFUSED: "bg-red-500",
+  },
+  text: {
+    WAITING_VALIDATION: "text-sky-600",
+    WAITING_CORRECTION: "text-white",
+    VALIDATED: "text-white",
+    REFUSED: "text-white",
+  },
+};
+
```
Comment (nicobret) [inline app/src/scenes/militaryPreparation/components/DocumentsPM.jsx:24]: Utiliser la constante dans engagement.utils.js ?

File: app/src/scenes/phase2/views/TableauDeBord.jsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    className={`${className} md:ml-auto text-xs font-normal ${theme.background[young.statusMilitaryPreparationFiles]} ${
+      theme.text[young.statusMilitaryPreparationFiles]
+    } rounded-sm px-2 py-[2px] `}>
+    {translateStatusMilitaryPreparationFiles(young.statusMilitaryPreparationFiles)}
+  </div>
+);
+
+export default function View() {
+  const { young } = useAuth();
+  console.log(young);
```
Comment (nicobret) [inline app/src/scenes/phase2/views/TableauDeBord.jsx]:
```suggestion
```

File: app/src/scenes/phase2/views/TableauDeBord.jsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  const phase2NumberHoursDone = young.phase2NumberHoursDone || 0;
+  // const phase2NumberHoursDone = 0;
```
Comment (nicobret) [inline app/src/scenes/phase2/views/TableauDeBord.jsx]:
```suggestion
```

File: app/src/scenes/phase2/views/TableauDeBord.jsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  const equivalences = useQuery({ queryKey: ["equivalence"], queryFn: () => fetchEquivalences(young._id) });
+
+  const missionDoneCards =
+    applications.data &&
+    equivalences.data &&
+    [...applications.data, ...equivalences.data]
+      .sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt))
+      .filter(({ status }) =>
+        [APPLICATION_STATUS.DONE, EQUIVALENCE_STATUS.VALIDATED, EQUIVALENCE_STATUS.WAITING_CORRECTION, EQUIVALENCE_STATUS.WAITING_VERIFICATION].includes(status),
+      );
```
Comment (nicobret) [inline app/src/scenes/phase2/views/TableauDeBord.jsx]: On affiche toutes les cartes, pas besoin de filter.

File: app/src/scenes/phase2/views/TableauDeBord.jsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  const missionCandidateCards =
+    applications.data && applications.data.sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt)).filter(({ status }) => status !== APPLICATION_STATUS.DONE);
+
+  if (young.statusPhase2 === YOUNG_STATUS_PHASE2.VALIDATED) return <Validated></Validated>;
```
Comment (nicobret) [inline app/src/scenes/phase2/views/TableauDeBord.jsx]: On supprime ce composant, il est remplacé par la home phase 2 déjà validée.

File: app/src/scenes/phase2/views/TableauDeBord.jsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                <HiPlus className="inline-block mr-2 text-xl align-text-bottom" />
+                Ajouter un engagement réalisé
+              </Link>
+            </div>
+          </span>
+        )}
+      </div>
+
+      {equivalences.isError ||
+        (applications.isError && (
```
Comment (nicobret) [inline app/src/scenes/phase2/views/TableauDeBord.jsx:88]: Ajouter un loader pour isPending ou pas besoin ?

**Review by nicobret — APPROVED** [review]

File: app/src/scenes/phase2/views/TableauDeBord.jsx (hunk 4 above)
Comment (marin42) [reply app/src/scenes/phase2/views/TableauDeBord.jsx]: On affiche pas les candidatures elles sont dans une autre partie :)

**Review by nicobret — APPROVED** [review]

**Pull request merged**